From 60ccecc7029a547fb455e01b83599fb12442b9e9 Mon Sep 17 00:00:00 2001
From: Lukasz Majczak <lukaszm@marvell.com>
Date: Fri, 14 Jun 2019 09:52:44 +0200
Subject: [PATCH 340/345] mtd: m25p80: Add Micron (MT25Q*) SPI flash devices.

Some Octeon73xx-based platforms failed to initialize the SPI.
Fix it by adding missing Micron SPI flash device ID's to the relevant drivers.

Change-Id: If5761dd2c66ab45e16a3fa319d50a634c14fbc44
Signed-off-by: Lukasz Majczak <lukaszm@marvell.com>
---
 drivers/mtd/devices/m25p80.c  | 12 ++++++++++++
 drivers/mtd/spi-nor/spi-nor.c | 12 ++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
index 24e1ea36dd2e..6dab668ae317 100644
--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -354,6 +354,18 @@ static const struct spi_device_id m25p_ids[] = {
 	{"m25p64"},	{"m25p128"},
 	{"w25x80"},	{"w25x32"},	{"w25q32"},	{"w25q32dw"},
 	{"w25q80bl"},	{"w25q128"},	{"w25q256"},
+	{ "mt25ql064" },
+	{ "mt25qu064" },
+	{ "mt25ql128" },
+	{ "mt25qu128" },
+	{ "mt25ql256" },
+	{ "mt25qu256" },
+	{ "mt25ql512" },
+	{ "mt25qu512" },
+	{ "mt25ql01g" },
+	{ "mt25qu01g" },
+	{ "mt25ql02g" },
+	{ "mt25qu02g" },
 
 	/* Flashes that can't be detected using JEDEC */
 	{"m25p05-nonjedec"},	{"m25p10-nonjedec"},	{"m25p20-nonjedec"},
diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 917db47f5148..b4ba0276de47 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -1033,6 +1033,18 @@ static const struct flash_info spi_nor_ids[] = {
 	{ "mx66l1g55g",  INFO(0xc2261b, 0, 64 * 1024, 2048, SPI_NOR_QUAD_READ) },
 
 	/* Micron */
+	{ "mt25ql064",   INFO(0x20ba17, 0, 64 * 1024,  128, SECT_4K) },
+	{ "mt25qu064",   INFO(0x20bb17, 0, 64 * 1024,  128, SECT_4K) },
+	{ "mt25ql128",   INFO(0x20ba18, 0, 64 * 1024,  256, SECT_4K) },
+	{ "mt25qu128",   INFO(0x20bb18, 0, 64 * 1024,  256, SECT_4K) },
+	{ "mt25ql256",   INFO(0x20ba19, 0, 64 * 1024,  512, SECT_4K) },
+	{ "mt25qu256",   INFO(0x20bb19, 0, 64 * 1024,  512, SECT_4K) },
+	{ "mt25ql512",   INFO(0x20ba20, 0, 64 * 1024, 1024, SECT_4K) },
+	{ "mt25qu512",   INFO(0x20bb20, 0, 64 * 1024, 1024, SECT_4K) },
+	{ "mt25ql01g",   INFO(0x20ba21, 0, 64 * 1024, 2048, SECT_4K) },
+	{ "mt25qu01g",   INFO(0x20bb21, 0, 64 * 1024, 2048, SECT_4K) },
+	{ "mt25ql02g",   INFO(0x20ba22, 0, 64 * 1024, 4096, SECT_4K) },
+	{ "mt25qu02g",   INFO(0x20bb22, 0, 64 * 1024, 4096, SECT_4K) },
 	{ "n25q016a",	 INFO(0x20bb15, 0, 64 * 1024,   32, SECT_4K | SPI_NOR_QUAD_READ) },
 	{ "n25q032",	 INFO(0x20ba16, 0, 64 * 1024,   64, SPI_NOR_QUAD_READ) },
 	{ "n25q032a",	 INFO(0x20bb16, 0, 64 * 1024,   64, SPI_NOR_QUAD_READ) },
-- 
2.25.1

